pkg/types/ccipocr3/generic_types.go

```diff
@@ -83,18 +83,26 @@ func (c ChainSelector) String() string {
 }
 
 type CCIPMsg struct {
-	ChainFeeLimit   BigInt        `json:"chainFeeLimit"`
-	Nonce           uint64        `json:"nonce"`
-	Sender          types.Account `json:"sender"`
-	Receiver        types.Account `json:"receiver"`
-	Strict          bool          `json:"strict"`
-	FeeToken        types.Account `json:"feeToken"`
-	FeeTokenAmount  BigInt        `json:"feeTokenAmount"`
-	Data            []byte        `json:"data"`
-	TokenAmounts    []TokenAmount `json:"tokenAmounts"`
-	SourceTokenData [][]byte      `json:"sourceTokenData"`
-
 	CCIPMsgBaseDetails
+	ChainFeeLimit   BigInt          `json:"chainFeeLimit"`
+	Nonce           uint64          `json:"nonce"`
+	Sender          types.Account   `json:"sender"`
+	Receiver        types.Account   `json:"receiver"`
+	Strict          bool            `json:"strict"`
+	FeeToken        types.Account   `json:"feeToken"`
+	FeeTokenAmount  BigInt          `json:"feeTokenAmount"`
+	Data            []byte          `json:"data"`
+	TokenAmounts    []TokenAmount   `json:"tokenAmounts"`
+	SourceTokenData [][]byte        `json:"sourceTokenData"`
+	Metadata        CCIPMsgMetadata `json:"metadata"`
+}
+
+type CCIPMsgMetadata struct {
+	// Version of the message metadata. Required in order to be able to parse the metadata
+	// by the underlying implementation.
+	Version uint64 `json:"version"`
```

Comment:
How is this version set / managed? Is this the CCIP capability semantic version?

Reply:
We shouldn't have different versions from the ccip capability version. I think its fine to use that here but we shouldn't create another set of versions